hexa/core/models/soft_delete.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from django.db import models
+from django.utils import timezone
+
+
+class DefaultSoftDeletedManager(models.Manager):
+    """
+    This manager will return all objects no marked as  soft deleted
+    """
+
+    def get_queryset(self):
```

Comment:
I would still create a `SoftDeleteQuerySet` to handle the bulk deletion (by default, it will delete the rows from DB and not soft delete them)